lib/zipkin/samplers/probabilistic.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Zipkin
+  module Samplers
+    # Probabilistic sampler
+    #
+    # Sample a portion of traces using trace_id as the random decision
+    class Probabilistic
+      def initialize(rate: 0.001)
+        if rate < 0.0 || rate > 1.0
+          raise "Sampling rate must be between 0.0 and 1.0, got #{rate.inspect}"
+        end
+        @boundary = TraceId::TRACE_ID_UPPER_BOUND * rate
+      end
+
+      def sample?(trace_id)
```

Comment:
Can a trace be accessed here via `trace_id`? If not, then the interface doesn't support smarter samplers that need to inspect the trace. I can imagine there's use for samplers that e.g. always allow traces that aren't similar to any other traces to get maximal coverage.

Reply:
No it cannot. Sampling is decided before any of the spans are created. go injects two parameters `trace_id` and `operation name` https://github.com/jaegertracing/jaeger-client-go/blob/master/sampler.go#L43 It seems that what you're describing is similar to this https://github.com/jaegertracing/jaeger-client-go/blob/master/sampler.go#L198

Reply:
Is it okay that you'll break compatibility with old samplers when you want to add `operatrion_name` as a parameter? If not, `trace_id` should be named variable.

Reply:
changed to named argument